koschei/backend/__init__.py

```diff
@@ -112,12 +116,17 @@ def submit_build(session, package):
     )
     if srpm_res:
         srpm, srpm_url = srpm_res
+        if session.build_from_repo_id:
+            target = None
+            build_opts.update({'repo_id': package.collection.latest_repo_id})
```

Comment:
You should set `build.repo_id`, otherwise you lose the benefits this feature may have